### PR TITLE
Remove auth+http clients in favor of gordon-gcp

### PR DESCRIPTION
**Pull request by econchick**
**What this PR does / why we need it**:

Since we [merged](https://github.com/spotify/gordon-gcp/pull/5) the http and auth clients into `gordon-gcp`, this PR removes the duplicated code and makes use of that package.

**Which issue(s) this PR fixes**

N/A

**Special notes for your reviewer**:

When I run `tox` locally, it b0rks for some reason, but works fine if I run the commands directly within a virtualenv. I'm hoping that Travis's running of the tests will restore my sanity...

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by econchick** [comment]
now that I've rebased, would someone mind reviewing? cc @spotify/alf

File: src/gordon_janitor_gcp/clients/_utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
```
Comment (descent3me) [inline src/gordon_janitor_gcp/clients/_utils.py:1]: Should we name a module that will be imported with a leading underscore?  I thought the underscore was to indicate it was private[-ish]?

**Review by descent3me — COMMENTED** [review]
A question around naming, but generally a fan of deleting stuff 👍

File: src/gordon_janitor_gcp/clients/_utils.py (hunk 1 above)
Comment (econchick) [reply src/gordon_janitor_gcp/clients/_utils.py:1]: wow delayed response! dang, how did I miss this.

You're right that it's to denote it's not a public module. But more so from the perspective of depending packages, i.e. in extending `gordon_janitor_gcp` for our Spotify-specific stuff, we should not import/use this `_utils` module. It's meant to only be called/used internally within this package (much like how semi private helper functions can still be called from public ones, just not externally). Does that make sense?

**Comment by econchick** [comment]
I'm going to cut a release after #18 is merged in.

**Pull request merged**